### PR TITLE
release: Remove gitian

### PR DESCRIPTION
**Pull request by jamescowens**
This commit removes gitian. We have not, and do not intend to, use gitian for releases. See https://github.com/bitcoin/bitcoin/pull/22418.

### CODE REVIEW

File: contrib/devtools/README.md [hunk 1]
```diff
@@ -97,18 +97,14 @@ Perform basic security checks on a series of executables.
 symbol-check.py
 ===============
 
-A script to check that the executables produced by gitian only contain
-certain symbols and are only linked against allowed libraries.
+A script to check that release executables only contain ceertain symbols and are only linked against
```
Comment (barton2526) [inline contrib/devtools/README.md]:
```suggestion
A script to check that release executables only contain certain symbols and are only linked against
```

**Review by barton2526 — APPROVED** [review]

**Comment by div72** [comment]
@jamescowens Could you retain the macdeploy docs and only change the Gitian part to Guix? Sure, we currently do not do deterministic builds nor sign them but that is a nice goal for the future in my opinion.

**Comment by div72** [comment]
On a related note, could you also delete gridcoin-community/gitian.sigs repo?

**Comment by jamescowens** [comment]
I will restore the verbiage, but indicate that at the current time we have not implemented Guix...

**Comment by jamescowens** [comment]
delete gridcoin-community/gitian.sigs repo? ... Done.

**Comment by jamescowens** [comment]
@div72 Ping.

**Comment by div72** [comment]
@jamescowens
```
$ git grep -l -i -r gitian .
ci/test/00_setup_env_win32.sh
contrib/devtools/symbol-check.py
contrib/macdeploy/README.md
doc/README.md
doc/release-process.md
```

**Comment by jamescowens** [comment]
The macdeploy/README.md is going to have a hit, see the commentary.

**Comment by jamescowens** [comment]
I think we will leave symbol-check.py in place. It will need to be updated, but that is NOT in scope of this PR.

**Review by div72 — APPROVED** [review]
utACK.

Related to #2249: Pushing a tag will automatically start the compilation of binaries, but it might also create a release owned by github-actions from the same tag. You providing an access token using secrets might work but you'll still have to edit the new popped release instead of making one. That issue is caused by our current action using `softprops/action-gh-release` but I am not versed enough in JS to make a replacement. Comments?

**Comment by jamescowens** [comment]
Hmm... good point. Let me think about that a bit. I will go ahead and merge this one though.

**Pull request merged**